docs/CONTRIBUTING.md

```diff
@@ -5,67 +5,133 @@ to make via issue, before making a change. Please note we have a
+
+  ```sh
+  asdf install scarb 0.7.0
+  asdf install scarb 2.6.4
+  ```
+
+- Install the two VSCode extensions for Cairo:
+  - [Cairo Syntax](https://marketplace.visualstudio.com/items?itemName=starkware.cairo)
+  - [Cairo Language Server](https://marketplace.visualstudio.com/items?itemName=ericglau.cairo-ls)
+
+## Setting up the environment
 
-#### Install foundry
+To set up a development environment, please follow these steps:
 
-To assess the soundness of our EVM, we compile common Solidity contracts and run
-tests on them through Kakarot.
+1. Clone the repo and navigate to it
 
-To be able to verify and compare the ABI and bytecode of the Solidity test
-contracts, first make sure you have
-[foundry installed on your machine](https://book.getfoundry.sh/getting-started/installation).
+   ```sh
+   git clone https://github.com/kkrt-labs/kakarot && cd kakarot
+   ```
 
-Then, run:
+2. Create and activate a virtual environment specifically for your Python
+   project by running `poetry shell`.
 
-`make build-sol`
+3. Install dependencies
 
-Common caveats:
+   ```sh
+   make setup
+   ```
 
-- python3.10 is not compatible with the cairo-lang library. Make sure poetry and
-  your pyenv are using the 3.9 version of Python. Your machine may have
-  conflicting versions of python.
-  - 3.9-dev will fail when running `make setup` since it is evaluated as 3.9.10+
-    and therefore an invalid PEP 440 version (Poetry
-    [enforces PEP 440 versioning](https://python-poetry.org/docs/faq#why-does-poetry-enforce-pep-440-versions)).
-- Mac M1 chips are subject to some quirks/bugs with regards to some
-  cryptographic libraries used by `cairo-lang`.
-  - you may need to run `brew install gmp`.
-  - if some c-compiler errors persist, refer to
-    [this Cairo issue for solutions](https://github.com/OpenZeppelin/nile/issues/22).
+4. Install katana
 
-To set up a development environment, please follow these steps:
+   ```sh
+   make install-katana
+   ```
 
-1. Clone the repo
+5. Build the Solidity contracts
 
    ```sh
-   git clone https://github.com/kkrt-labs/kakarot
+   make build-sol
    ```
 
-2. Install dependencies
+6. Copy the default environment variables
 
    ```sh
-   make setup
+   cp .env.example .env
+   ```
+
+7. Start a local katana instance
+
+   ```sh
+   make run-katana
    ```
 
-3. Run tests
+8. Run tests
 
    ```sh
    make test
    ```
 
+Common caveats:
+
+- You will need to symlink `starknet-compile-deprecated` (new name of the
+  starknet-compile binary) to `starknet-compile` in order to make the CairoLS
+  VSCode extension work.
+
+  ```bash
+  ln -s <YOUR_PATH_TO_YOUR_PYTHON_VENV_BINARIES>/starknet-compile-deprecated <YOUR_PATH_TO_LOCAL_BINARIES>/starknet-compile
+  # example: ln -s /Users/eliastazartes/code/kakarot/.venv/bin/starknet-compile-deprecated /usr/local/bin/starknet-compile
+  ```
+
+  If you followed the instructions above, you can run
+  `poetry show cairo-lang -v` to find the path to your cairo-lang dependency,
```

Comment:
starknet-compile and not cairo-lang ?

Reply:
no, `cairo-lang` shows where the package is installed, and the `starknet-compile` binary will be under `bin/starknet-compile-deprecated`